src/paths.jl

```diff
@@ -0,0 +1,36 @@
+"""
+
+Draws lines segments connecting input points.
+Parameters:
+img: 2d array
+vertices: coordinates of points as array of CartesianIndex
+color (optional): color of line segments
+closed (keyword): whether to connect first and last points
+
+In case a point is out-of-bounds, it throws an error after drawing the line segments till that point.
+"""
+
+path{T<:Colorant}(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1}, color::T=one(T); closed::Bool=true) = path!(copy(img),vertices,color,closed=closed)
+
+function path!{T<:Colorant}(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1}, color::T=one(T); closed::Bool=true)
```

Comment:
Those are good thoughts. I'd say, let's use the API we want and then change later if we discover the performance isn't acceptable. By the time someone notices a problem, one can always hope that the penalty for keyword arguments will be gone anyway.